File: taurus-protect-sdk-typescript/src/models/fee.ts
/**
 * Fee models for Taurus-PROTECT SDK.
 *
 * Provides models for network fee information retrieved from the API.
 */

import type { Currency } from './currency';

/**
 * Represents a network fee entry with currency information (v2 API format).
 *
 * Provides detailed fee information including currency metadata.
 */
export interface FeeV2 {
  /** Currency identifier */
  readonly currencyId?: string;
  /** Fee value (amount) */
  readonly value?: string;
  /** Denomination (e.g., "wei", "satoshi") */
  readonly denom?: string;
  /** Currency information */
  readonly currencyInfo?: Currency;
  /** Last update timestamp */
  readonly updateDate?: Date;
}

File: taurus-protect-sdk-go/pkg/protect/model/fee.go
package model

import "time"

// FeeV2 represents a native currency fee estimate with detailed information.
// This is returned by the GetFeesV2 endpoint.
type FeeV2 struct {
	// CurrencyID is the unique identifier of the native currency.
	CurrencyID string `json:"currency_id,omitempty"`
	// Value is the fee amount in the native currency's smallest unit.
	Value string `json:"value,omitempty"`
	// Denom is the denomination/unit of the fee value.
	Denom string `json:"denom,omitempty"`
	// CurrencyInfo contains detailed information about the native currency.
	CurrencyInfo *CurrencyInfo `json:"currency_info,omitempty"`
	// UpdateDate is when this fee estimate was last updated.
	UpdateDate time.Time `json:"update_date,omitempty"`
}

// GetFeesV2Result contains the result of the GetFeesV2 call.
type GetFeesV2Result struct {
	// Fees is the list of native currency fee estimates.
	Fees []*FeeV2 `json:"fees"`
}
